graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt

```diff
@@ -113,4 +130,32 @@ class DataFetcherInvoker internal constructor(
         return "Could not resolve parameter [${param.parameterIndex}] in " +
             param.executable.toGenericString() + if (message.isNotEmpty()) ": $message" else ""
     }
+
+    /**
+     * Wrap the call to a data fetcher in CompletableFuture to enable parallel behavior.
+     * Used when virtual threads are enabled.
+     */
+    private fun wrapInCompletableFuture(function: () -> Any?): Any? {
+        return CompletableFuture.supplyAsync({
+            return@supplyAsync function.invoke()
+        }, taskExecutor)
+    }
+
+    /**
+     * Decides if a data fetcher method should be wrapped in CompletableFuture automatically.
+     * This is only done when a taskExecutor is available, and if the data fetcher doesn't explicitly return CompletableFuture already.
+     * Used when virtual threads are enabled.
+     */
+    private fun shouldWrapInCompletableFuture(kFunc: KFunction<*>): Boolean {
+        return taskExecutor != null && !kFunc.returnType.isSubtypeOf(typeOf<CompletionStage<Any>>())
```

Comment:
This condition covers CompletableFuture as well?

Reply:
yes, CompletionStage sits at the top of that type structure